tests/framework/clients/rancher/v1/client.go

```diff
@@ -164,12 +169,29 @@ func (c *SteveClient) Replace(obj *SteveAPIObject) (*SteveAPIObject, error) {
 }
 
 func (c *SteveClient) List(query url.Values) (*SteveCollection, error) {
+	// sometimes it is necessary to retry the GetCollectionURL due to the schema not being updated
+	// fast enough after a cluster has been provisioned
+	var backoff = wait.Backoff{
+		Duration: duration,
+		Factor:   factor,
+		Jitter:   0,
+		Steps:    steps,
+	}
 	resp := &SteveCollection{}
 	var jsonResp map[string]any
-	url, err := c.apiClient.Ops.GetCollectionURL(c.steveType, "GET")
+	var url string
+	var getCollectionURL error
+	err := wait.ExponentialBackoff(backoff, func() (done bool, err error) {
+		url, getCollectionURL = c.apiClient.Ops.GetCollectionURL(c.steveType, "GET")
```

Comment:
`GetCollectionURL` uses `APIOperations.Types` to get the schema, which I think is what would be empty. What is supposed to populate it while this backoff is running?

Would it make any sense to add retries to https://github.com/rancher/rancher/blob/3f34c0bc977978f1a8c18cd873e66d3254363e50/tests/framework/pkg/clientbase/common.go#L169 so that it doesn't return a client until the schemas are ready?

Reply:
Putting it in the proxy  client sounds good :+1:

Reply:
Awesome I will update it, thank you

Reply:
@cmurphy so I ended up checking a limit which amounts to 1037 schemas. As it's more of case of it not having all the schemas populated rather than it being zero, I believe. Let me know what you think.

Reply:
never mind this check won't work. I'm push up a fix shortly

Reply:
So I changed the logic, when instantiating a new ProxyClient, I'll create at least twice to make sure the number of schemas stays the across the different instantiations to ensure everything is available. Let me know what you think.

Reply:
It won't necessarily be exactly 1037. But if it's >0, I would think at a minimum pods would be in that list to start, so checking for >0 would be good enough for that case.  Is it mainly basic resources like pods that are the problem, or are other resource clients failing to get their schema? Other things like metrics or rancher CRDs might take longer to populate, maybe even seconds, so checking that the size of the schema is the same after two passes might not guarantee that it's finished populating.

Reply:
Ok I'll make the change. We've really only seen this with pods, and believe with `management.cattle.io.setting` one time. But I agree, I'll make that change.

Reply:
Done